src/nupic/research/spatial_pooler.py

```diff
@@ -1259,25 +1260,28 @@ def _mapPotential(self, index, wrapAround=False):
     @param wrapAround: A boolean value indicating that boundaries should be
                     fignored.
     """
-    index = self._mapColumn(index)
-    indices = self._getNeighborsND(index,
-                                   self._inputDimensions,
-                                   self._potentialRadius,
-                                   wrapAround=wrapAround)
-    indices.append(index)
-    indices = numpy.array(indices, dtype=uintType)
 
-    # TODO: See https://github.com/numenta/nupic.core/issues/128
-    indices.sort()
+    centerInput = self._mapColumn(index)
+
+    if wrapAround:
+      columnInputs = topology.wrappingNeighborhood(centerInput,
+                                                   self._potentialRadius,
+                                                   self._inputDimensions)
+    else:
+      columnInputs = topology.neighborhood(centerInput,
+                                           self._potentialRadius,
+                                           self._inputDimensions)
+
+    columnInputs = columnInputs.astype(uintType)
 
     # Select a subset of the receptive field to serve as the
     # the potential pool
-    numPotential = int(round(indices.size * self._potentialPct))
-    selectedIndices = numpy.empty(numPotential, dtype=uintType)
-    self._random.sample(indices, selectedIndices)
+    numPotential = int(round(columnInputs.size * self._potentialPct))
```

Comment:
Fixed. Thanks, I didn't know about that. I do that elsewhere because the existing code already did it.